Source/Scene/Model.js

```diff
@@ -1453,6 +1453,103 @@ define([
+                                            for (var varName in technique.attributes) {
+                                                if (technique.attributes.hasOwnProperty(varName)) {
+                                                    if (technique.attributes[varName] === attributeName) {
+                                                        attributeVarName = varName;
+                                                        break;
+                                                    }
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                                // Skip if the attribute is not used by the material, e.g., because the asset was exported
+                                // with an attribute that wasn't used and the asset wasn't optimized.
+                                if (defined(attributeVarName)) {
+                                    var a = accessors[primitiveAttributes[attributeSemantic]];
+                                    if (a.hasOwnProperty('extensions')) {
+                                        var extensions = a.extensions;
+                                        if (extensions.hasOwnProperty('WEB3D_quantized_attributes')) {
+                                            var quantizedAttributes = extensions.WEB3D_quantized_attributes;
+                                            var decodeMatrix = quantizedAttributes.decodeMatrix;
+                                            decodeMatrix = decodeMatrix.map(function(val) {
+                                                return val.toFixed(16);
+                                            });
+
+                                            var newMain = "decoded_" + attributeSemantic;
+                                            var matrixName = "decode_" + attributeSemantic + "_matrix";
+                                            var decodedAttributeVarName = attributeVarName.replace('a_', 'dec_');
+
+                                            // replace usages of the original attribute with the decoded version, but not the declaration
+                                            var first = true;
+                                            shader = shader.replace(new RegExp(attributeVarName, 'g'), function(match) {
+                                                if (first) {
+                                                    first = false;
+                                                    return match;
+                                                }
+                                                return decodedAttributeVarName;
+                                            });
+                                            // declare decoded attribute
+                                            shader = "vec3 " + decodedAttributeVarName + ";\n" + shader;
```

Comment:
Anything can be listed as an attribute for a mesh could be used. According to the spec that is: `POSITION`, `NORMAL`, `TEXCOORD`, `COLOR`, `JOINT`, `JOINTMATRIX`, and `WEIGHT`. If we decide this is the correct approach, I will implement the other data types.